engine/src/test/java/io/graphqlcrud/SQLDataFetcherTest.java

```diff
@@ -195,6 +195,179 @@ public void testDeepNestedQuery() throws Exception {
+        String query = "{\n" +
+                "  customers (filter: {\n" +
+                "    and: {\n" +
+                "      LASTNAME: {\n" +
+                "        eq: \"Smith\"\n" +
+                "      },\n" +
+                "      or: {\n" +
+                "        LASTNAME: {\n" +
+                "          eq: \"Doe\"\n" +
+                "        }\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }) {\n" +
+                "    SSN\n" +
+                "    FIRSTNAME\n" +
+                "  }\n" +
+                "}";
+        String result = executeSQL(query);
+        String expected = "select\n" +
+                "  \"g0\".\"SSN\" \"SSN\",\n" +
+                "  \"g0\".\"FIRSTNAME\" \"FIRSTNAME\"\n" +
+                "from PUBLIC.CUSTOMER \"g0\"\n" +
+                "where (\n" +
+                "  \"g0\".\"LASTNAME\" = 'Smith'\n" +
+                "  or \"g0\".\"LASTNAME\" = 'Doe'\n" +
+                ")\n" +
+                "order by \"g0\".\"SSN\"";
+        Assertions.assertEquals(expected,result);
+    }
+
+    @Test
+    public void deepNestedFilterQuery() throws Exception {
+        String query = "{\n" +
+                "  customers (filter: {\n" +
+                "    or: {\n" +
+                "      LASTNAME: {\n" +
+                "        eq: \"Smith\"\n" +
+                "      },\n" +
+                "      FIRSTNAME: {\n" +
+                "        eq: \"John\"\n" +
+                "      },\n" +
+                "      and: {\n" +
+                "        SSN: {\n" +
+                "          eq: \"CST01002\"\n" +
+                "        }\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }) {\n" +
+                "    SSN\n" +
+                "    LASTNAME\n" +
+                "    accounts (filter: {\n" +
+                "      and: {\n" +
+                "        ACCOUNT_ID: {\n" +
+                "          eq: 19980001\n" +
+                "        }\n" +
+                "      }\n" +
+                "    }) {\n" +
+                "      ACCOUNT_ID\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+        String result = executeSQL(query);
+        String expected = "select\n" +
+                "  \"g0\".\"SSN\" \"SSN\",\n" +
+                "  \"g0\".\"LASTNAME\" \"LASTNAME\",\n" +
+                "  (\n" +
+                "    select json_arrayagg(json_object(key 'ACCOUNT_ID' value \"g1\".\"ACCOUNT_ID\"))\n" +
+                "    from PUBLIC.ACCOUNT \"g1\"\n" +
+                "    where (\n" +
+                "      \"g0\".\"SSN\" = \"g1\".\"SSN\"\n" +
+                "      and \"g1\".\"ACCOUNT_ID\" = 19980001\n" +
+                "    )\n" +
+                "  ) \"accounts\"\n" +
+                "from PUBLIC.CUSTOMER \"g0\"\n" +
+                "where (\n" +
+                "  (\n" +
```

Comment:
The result of this is not correct. it should be like ```lastname = 'smith' and firstname='John' and ssn = 'cst01002'``` a `or` as enclosing does not mean the attributes inside are separated by  `or`. `or` is always adjunct operation where you could have said

```
{
  customers (filter: {
    LASTNAME: {
      eq: "Smith"
    },
    FIRSTNAME: {
      eq: "John"
    },
    or: {
      SSN: {
        eq: "CST01002"
      }
    }
  }
}
```

which should result in `(lastname = 'smith' and firstname='John' ) or ssn = 'cst01002'` 

overall what I am saying `or` should not have had any effect on the query above

Reply:
Thanks for pointing out that, I will correct it.

Reply:
We found ourselves looking at the semantics of `or` operator recently in Graphback

/cc @craicoverflow if you want to share your experience.

Reply:
yeah, We checked out competitors (AppSync) and their query output match that by @rareddy above.

Reply:
@ani-sha did you make sure the previous original test as is should have behaved the way I listed above, as the test is not wrong the results were

Reply:
@ani-sha keep the test case as is from before, make sure the results were correct.